vault/data_source_azure_access_credentials.go

```diff
@@ -134,19 +146,25 @@ func azureAccessCredentialsDataSourceRead(d *schema.ResourceData, meta interface
 	log.Printf("[DEBUG] Successfully read %q from Vault", configPath)
 
 	subscriptionID := ""
-	if subscriptionIDIfc, ok := config.Data["subscription_id"]; ok {
+	if v, ok := d.GetOk("subscription_id"); ok {
```

Comment:
Can you set line 142 that if subscription id and tenant id are defined in the provider then it doesn't need to do the vault config read at all? I don't believe it's used anywhere else in the code and would mean the token that is being used by terraform can have fewer permissions in vault as it no longer needs to read the config.

Reply:
Fixed in https://github.com/hashicorp/terraform-provider-vault/pull/1384/commits/bcb6464a0fbda175162d7cb9b9cbde4bb83cf781